packages/components/src/components/text/text.scss

```diff
@@ -0,0 +1,27 @@
+.text {
+    @each $name, $size in $FONT_SIZES {
+        &--#{$name} {
+            font-size: $size;
+        }
+    }
+    @each $name, $color in $FONT_COLORS {
+        &--#{$name} {
+            color: $color;
+        }
+    }
+    @each $name, $weight in $FONT_WEIGHTS {
+        &--#{$name} {
+            font-weight: $weight;
+        }
+    }
+    @each $name, $align in $TEXT_ALIGN {
+        &--#{$name} {
+            text-align: $align;
+        }
+    }
+    @each $name, $lineHeight in $LINEHEIGHTS {
+        &__line-height--#{$name} {
+            line-height: $lineHeight;
+        }
+    }
```

Comment:
(throughout) Iterating over basic sass variables and generating class names based on that will make the output size a bit bloated. It is better to use variables, and then reference the appropriate one in the JSX component.